src/github.com/matrix-org/dendrite/roomserver/input/state.go

```diff
@@ -114,8 +116,94 @@ func calculateAndStoreStateAfterManyEvents(db RoomEventDatabase, roomNID types.R
 	return db.AddState(roomNID, nil, state)
 }
 
+// loadStateEvents loads the matrix events for a list of state entries.
+// Returns a list of events and a map from string event ID back to state entry.
```

Comment:
> Returns a list of events

I can tell that from the function definition. Tell me something the function definition can't say:
 - What is this list sorted by?
 - Is it a guaranteed 1:1 mapping from `[]types.StateEntry`?
 - Are the events all state events?